sharding/nodesSetup_test.go

```diff
@@ -514,3 +726,503 @@ func TestNodesSetup_NodesAboveMaxShardLimit(t *testing.T) {
+			Address: address[1],
+		},
+		{
+			PubKey:  pubKeys[2],
+			Address: address[2],
+		},
+	}
+
+	var listOfExpectedConvertedPubKeysEligibleNodes = make([]string, 2)
+	pubKeyConverter := mock.NewPubkeyConverterMock(96)
+
+	for i := 0; i < 2; i++ {
+		convertedValue, err := pubKeyConverter.Decode(listOfInitialNodes[i].PubKey)
+		require.Nil(t, err)
+		require.NotNil(t, convertedValue)
+		listOfExpectedConvertedPubKeysEligibleNodes[i] = string(convertedValue)
+	}
+
+	ns, err := createTestNodesSetup(argsTestNodesSetup{
+		shardConsensusSize: 1,
+		shardMinNodes:      1,
+		metaConsensusSize:  1,
+		metaMinNodes:       1,
+		numInitialNodes:    3,
+		genesisMaxShards:   1,
+	})
+	require.Nil(t, err)
+	ns.Hysteresis = 0.2
+	ns.Adaptivity = false
+
+	ns = createAndAssignNodes(ns, noOfInitialNodes)
+
+	allEligibleNodes := ns.InitialNodesPubKeys()
+
+	require.Equal(t, listOfExpectedConvertedPubKeysEligibleNodes[0], allEligibleNodes[(core.MetachainShardId)][0])
+	require.Equal(t, listOfExpectedConvertedPubKeysEligibleNodes[1], allEligibleNodes[0][0])
+
+}
+
+func TestNodesSetup_InitialEligibleNodesPubKeysForShardShouldErrShardIdOutOfRange(t *testing.T) {
+	t.Parallel()
+
+	noOfInitialNodes := 3
+
+	ns, err := createTestNodesSetup(argsTestNodesSetup{
+		shardConsensusSize: 1,
+		shardMinNodes:      1,
+		metaConsensusSize:  1,
+		metaMinNodes:       1,
+		numInitialNodes:    3,
+		genesisMaxShards:   1,
+	})
+	require.Nil(t, err)
+	ns.Hysteresis = 0.2
+	ns.Adaptivity = false
+
+	ns = createAndAssignNodes(ns, noOfInitialNodes)
+
+	returnedPubKeys, err := ns.InitialEligibleNodesPubKeysForShard(1)
+	require.Nil(t, returnedPubKeys)
+	require.Equal(t, err, ErrShardIdOutOfRange)
+
+}
+
+func TestNodesSetup_InitialEligibleNodesPubKeysForShardShouldWork(t *testing.T) {
+	t.Parallel()
+
+	noOfInitialNodes := 3
+
+	var listOfInitialNodes = [3]InitialNode{
+		{
+			PubKey:  pubKeys[0],
+			Address: address[0],
+		},
+		{
+			PubKey:  pubKeys[1],
+			Address: address[1],
+		},
+		{
+			PubKey:  pubKeys[2],
+			Address: address[2],
+		},
+	}
+
+	var listOfExpectedPubKeysEligibleNodes = make([]string, 2)
+	pubKeyConverter := mock.NewPubkeyConverterMock(96)
+
+	for i := 0; i < 2; i++ {
+		convertedValue, err := pubKeyConverter.Decode(listOfInitialNodes[i].PubKey)
+		require.Nil(t, err)
+		require.NotNil(t, convertedValue)
+		listOfExpectedPubKeysEligibleNodes[i] = string(convertedValue)
+	}
+
+	ns, err := createTestNodesSetup(argsTestNodesSetup{
+		shardConsensusSize: 1,
+		shardMinNodes:      1,
+		metaConsensusSize:  1,
+		metaMinNodes:       1,
+		numInitialNodes:    3,
+		genesisMaxShards:   1,
+	})
+	require.Nil(t, err)
+	ns.Hysteresis = 0.2
+	ns.Adaptivity = false
+
+	ns = createAndAssignNodes(ns, noOfInitialNodes)
+
+	allEligibleNodes, err := ns.InitialEligibleNodesPubKeysForShard(0)
+
+	require.Nil(t, err)
+	require.Equal(t, listOfExpectedPubKeysEligibleNodes[1], allEligibleNodes[0])
+}
+
+func TestNodesSetup_NumberOfShardsShouldWork(t *testing.T) {
```

Comment:
all these getters, from L960-L1180 to could have been one single test, called TestNodesSetup_Getters, with only one call to createTestNodesSetup and then called all Getters

Reply:
updated